matscipy/surface_reconstruction.py

```diff
@@ -1,4 +1,5 @@
 import numpy as np
```

Comment:
This file has a horrendous number of DRY violations. It's because there's a lot of slightly different (but mostly the same) code when you're mapping 2x1 or 3x1 reconstructions. When I was writing it, my plan was to get it working, see what lines were in both files, and then try to generalise what I could. I never got round to the generalising step - but I will definitely come back to that at some point